html/functions.php

```diff
@@ -89,3 +69,56 @@ function createReplyTweet($text, $user_id, $reply_id)
     $stmt->bindValue(':reply_id', $reply_id, PDO::PARAM_INT);
     return $stmt->execute();
 }
+
+function createFavorite($member_id, $post_id)
+{
+    $sql = 'insert into favorites (member_id, post_id, created_at, updated_at)';
+    $sql .= ' values (:member_id, :post_id, :created_at, :updated_at)';
+    $now = date("Y-m-d H:i:s");
+    $stmt = getPdo()->prepare($sql);
+    $stmt->bindValue(':member_id', $member_id, PDO::PARAM_STR);
+    $stmt->bindValue(':post_id', $post_id, PDO::PARAM_INT);
+    $stmt->bindValue(':created_at', $now, PDO::PARAM_STR);
+    $stmt->bindValue(':updated_at', $now, PDO::PARAM_STR);
+    return $stmt->execute();
+}
+
+function deleteFavorite($member_id, $post_id)
+{
+    $sql = 'delete from favorites';
+    $sql .= ' where member_id = :member_id and post_id = :post_id';
+    $stmt = getPdo()->prepare($sql);
+    $stmt->bindValue(':member_id', $member_id, PDO::PARAM_INT);
+    $stmt->bindValue(':post_id', $post_id, PDO::PARAM_INT);
+    return $stmt->execute();
+}
+
+function isMyfavorite($member_id, $post_id)
+{
+    $sql = 'select post_id';
+    $sql .= ' from favorites';
+    $sql .= ' where member_id = :member_id and post_id = :post_id';
+    $sql .= ' order by updated_at desc';
+    $stmt = getPdo()->prepare($sql);
+    $stmt->bindValue(':member_id', $member_id, PDO::PARAM_INT);
+    $stmt->bindValue(':post_id', $post_id, PDO::PARAM_INT);
+    $stmt->execute();
+    $data = $stmt->fetchAll(PDO::FETCH_ASSOC);
+    $fabList = array();
+    foreach ($data as $d) {
+        $fabList = $d;
+    }
+    return $fabList;
```

Comment:
これだと $fabList 自体を foreach で毎回 $d で上書きしているので無意味な処理ですね！詰め替えがしたいなら
```php
$fabList[] = $d;
```
ですね。

しかし要はクエリ結果が1以上か否かが知りたいだけなので、$fabList をやめて次のように修正しましょう。

```suggestion
    return (0 < count($data));
```